### ISSUE TITLE
NVDA not inform about if an e-mail in Outlook 2013 is answered or forwarded

### ISSUE THREAD

**Issue by andersbp** [nvaccess/nvda#6911]
Hi everyone

If I want to read messages in Outlook 2013 and use the arrow keys to scroll up and down in my folders (Inbox, Sent Mail and more), unlike JAWS, NVDA does not announce if an e-mail is replied / forwarded.
Every time that NVDA comes to an e-mail that is either replied / forwarded, Nvda only announce that the e-mail have an attached file. The e-mails not has an attached file.

Is it an error in NVDA?
If yes, I will be glad for a correction.

Thanks in advance.

Regards,
Anders

**Comment by PratikP1** [comment]
This problem occurs in Outlook 2016 as well. Reply and forward flags are not announced. In Outlook 2016, the attachment issue occurs at the other end of the forward. For example, if I forward a message from one of my accounts to another, the received message shows that it has an attachment. Similarly, other forwarded messages that I’ve received show the same attachment announcement.

**Comment by feerrenrut** [comment]
Related to #4971 (status of is read, replied to, or forwarded) and #6374 (Outlook 2016 flagged items are not read in the message list)

**Comment by travisroth** [comment]
In Outlook 2016, the Replied and Forwarded status is in the value attribute of the list item UIA object which is being used in class UIAGridRow. UIAGridRow overrides value setting to Nothing so this is lost. If it could be added as an item before overriding value? (Note overriding value itself is still nice as it is to verbose and is announced at the end. I'd prefer to have Replied/Forwarded announced at the beginning. or have it user configurable.)
Unfortunately I do not know about earlier Outlooks. 
Travis

**Comment by josephsl** [comment]
Hi, alternatively, one can use table nav commands (Control+Alt+left/right arrows) to navigate to and announce this info, as it also inherits that power from NVDAObjects.behaviors.RowWithFakeNavigation. Thanks.

**Comment by josephsl** [comment]
Hi, I see what’s up… After replying to an earlier comment, I tried to look for reply/forward info but couldn’t find it (hoping it was one of the child names inside this row). Thanks.

**Comment by travisroth** [comment]
Hi,

I did not find it as a child. Only as the value of the list item and row navigation did not work for me.

As I said value is being overridden most likely to eliminate some verboseness, which I agree with. When I commented out that line (value=””) the value was announced at the end of the row announcement.

Hope this helps.

Travis

**Comment by josephsl** [comment]
Hi, I see. @MichaelDCurran, thoughts? Thanks.

**Comment by leonardder** [comment]
What is the current status of this issue? Honestly I thought it was fixed as part of #6374, but I made a wrong assumption there.

**Comment by ruifontes** [comment]
Hi Leonard!
The problem still exists.

Rui


-----Mensagem Original----- 
De: Leonard de Ruijter
Data: 1 de setembro de 2017 08:03
Para: nvaccess/nvda
Cc: Subscribed
Assunto: Re: [nvaccess/nvda] NVDA not inform about if an e-mail in Outlook 
2013 is answered or forwarded (#6911)



What is the current status of this issue? Honestly I thought it was fixed as 
part of #6374, but I made a wrong assumption there.

**Comment by feerrenrut** [comment]
@leonardder I'm not sure what's going on with this one, why is it labeled needs external fix? @Qchristensen believes that the replied status is reported with Narrator.

**Comment by michaelDCurran** [comment]
Replied / forwarded / read status is exposed by the control in plain 
language. It cannot be programmatically detected. Although Narrator 
announces this, it announces it in a non-configurable inefficient way.

We tried parsing the text to try to pull out replied / forwarded etc, 
but ran in to problems with several languages.

**Comment by mdyer1** [comment]
Not sure how this helps, but jaws 2018 speaks the status of a message that has been replied to.  Not sure how there doing this, but wanted to give my input on this since this is something people want.  Matthew

**Comment by leonardder** [comment]
JAWS' logic is based on the UIA value and language specific. When you use a Dutch Outlook with an English JAWS, it announces the read state for every read message, which is quite redundant IMO.

I proposed three possibilities in https://github.com/nvaccess/nvda/pull/8608#pullrequestreview-144384571

1. Revert the value = None for outlook 2013 and above. In that case, read/unread state will be announced after the current content.
2. Similar to 1, but map the value to the description object attribute. This allows people to toggle descriptions on and off in a per app profile, for example.
3. Add an object gesture to GridRow that reads the value.

Note that the value contains read/unread status, forwarded/replied as well as the message class (i.e. meeting request)
